### PR TITLE
Fixed upload does not exist error during image push operation.

### LINKED ISSUE #861: Upload does not exist during image push
```
pulp [None]: django.request:ERROR: Internal Server Error: /v2/pulp/pulp-operator/blobs/uploads/67467423-7b0c-4b5d-b6a4-44a99cd18355
  Traceback (most recent call last):
    File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 47, in inner
      response = get_response(request)
    File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 181, in _get_response
      response = wrapped_callback(request, *callback_args, **callback_kwargs)
    File "/usr/local/lib/python3.10/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
      return view_func(*args, **kwargs)
    File "/usr/local/lib/python3.10/site-packages/rest_framework/viewsets.py", line 125, in view
      return self.dispatch(request, *args, **kwargs)
    File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 509, in dispatch
      response = self.handle_exception(exc)
    File "/usr/local/lib/python3.10/site-packages/pulp_container/app/registry_api.py", line 321, in handle_exception
      response = super().handle_exception(exc)
    File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 469, in handle_exception
      self.raise_uncaught_exception(exc)
    File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
      raise exc
    File "/usr/local/lib/python3.10/site-packages/rest_framework/views.py", line 506, in dispatch
      response = handler(request, *args, **kwargs)
    File "/usr/local/lib/python3.10/site-packages/pulp_container/app/registry_api.py", line 688, in put
      raise e_upload
    File "/usr/local/lib/python3.10/site-packages/pulp_container/app/registry_api.py", line 678, in put
      upload = models.Upload.objects.get(pk=pk, repository=repository)
    File "/usr/local/lib/python3.10/site-packages/django/db/models/manager.py", line 85, in manager_method
      return getattr(self.get_queryset(), name)(*args, **kwargs)
    File "/usr/local/lib/python3.10/site-packages/django/db/models/query.py", line 435, in get
      raise self.model.DoesNotExist(
  pulp_container.app.models.Upload.DoesNotExist: Upload matching query does not exist.

```

### PR DESCRIPTION
**Pull request by ipanova**
closes #861

(cherry picked from commit e61849214b488fb32c293904503f804f0d0d7843)

### CODE REVIEW

File: pulp_container/app/registry_api.py [hunk 1]
```diff
@@ -645,10 +646,14 @@ def put(self, request, path, pk=None):
...
             # Upload has been deleted => task has started or even finished
-            task = Task.objects.filter(
-                name__endswith="add_and_remove",
-                reserved_resources_record__resource=f"upload:{pk}",
-            ).last()
+            task = (
+                Task.objects.filter(name__endswith="add_and_remove")
+                .filter(
+                    Q(reserved_resources_record__contains=[f"shared:upload:{pk}"])
+                    | Q(reserved_resources_record__contains=[f"upload:{pk}"])
```
Comment (mdellweg) [inline pulp_container/app/registry_api.py:653]: Is this really needed? Looking at
https://github.com/pulp/pulp_container/pull/868/files#diff-1b944e15118f82233aa03a435e9e05c188ad0d371b71be43260506b4a5eeddffL705
We never create the "shared:" version with this version of pulp_container.

Comment (ipanova) [reply pulp_container/app/registry_api.py:653]: 2.8 is compat with 3.14 and 3.15

Comment (ipanova) [reply pulp_container/app/registry_api.py:653]:
```
[ipanova@puffy pulpcore]$ git grep 'shared:'
pulpcore/tasking/pulpcore_worker.py:                    if not resource.startswith("shared:")
pulpcore/tasking/pulpcore_worker.py:                    if resource.startswith("shared:") and resource[7:] not in exclusive_resources
pulpcore/tasking/tasks.py:                (f"shared:{resource}" for resource in _validate_and_get_resources(shared_resources))
[ipanova@puffy pulpcore]$ git checkout 3.14
Switched to branch '3.14'
Your branch is up to date with 'upstream/3.14'.
[ipanova@puffy pulpcore]$ git grep 'shared:'
[ipanova@puffy pulpcore]$ 
```

Comment (mdellweg) [reply pulp_container/app/registry_api.py:653]: What i mean is that we declare the fake resource to be shared in the same file (at the dispatch some lines further down) [not on this branch]. And as long as we cannot _use_ shared resources on this branch for compatibility, we don't need to expect the fake resource to be listed shared, even if the version of pulpcore in use allows for it.

Comment (ipanova) [reply pulp_container/app/registry_api.py]: i see, yes you are right

**Comment by ipanova** [comment]
in that case this PR is unnecessary, thank for pointing that out

**Pull request closed without merging**